tests/e2e/performance/csi_tests/test_pvc_multi_snapshot_performance.py

```diff
@@ -1,42 +1,445 @@
+        except Exception as ex:
+            err_msg = f"Failed to create new snapshot class : {snapclass_name} [{ex}]"
+            log.error(err_msg)
+            raise Exception(err_msg)
+        return ocs_obj
+
+    def create_snapshot(self, snap_num):
+        """
+        Creating snapshot of volume, and measure the creation time
+
+        Args:
+            snap_num (int) the number of snapshot to create
+
+        Returns:
+            int: the creation time of the snapshot (in sec.)
+
+        """
+        log.info(f"Taking snapshot number {snap_num}")
+        # Getting UTC time before test starting for log retrieve
+        UTC_datetime = datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+        snap_name = f"pvc-snap-{snap_num}-"
+        snap_name += self.pvc_obj.name.split("-")[-1]
+        self.snap_templ["metadata"]["name"] = snap_name
+        self.snap_templ["spec"]["volumeSnapshotClassName"] = self.snap_class.name
+
+        fd, tmpfile = tempfile.mkstemp(suffix=".yaml", prefix="Snap")
+        log.debug(f"Going to create {tmpfile}")
+        with open(tmpfile, "w") as f:
+            yaml.dump(self.snap_templ, f, default_flow_style=False)
+
+        res = run_oc_command(cmd=f"create -f {tmpfile}", namespace=self.nss_name)
+        if ERRMSG in res[0]:
+            err_msg = f"Failed to create snapshot : {res}"
+            log.error(err_msg)
+            raise Exception(err_msg)
+
+        # wait until snapshot is ready
+        timeout = 600
+        sleep_time = 5
+
+        while timeout > 0:
+            res = run_oc_command(
+                f"get volumesnapshot {snap_name} -o yaml", namespace=self.nss_name
+            )
+
+            if ERRMSG not in res[0]:
+                res = yaml.safe_load("\n".join(res))
+                log.debug(f"The command output is : {yaml.dump(res)}")
+                try:
+                    if res["status"]["readyToUse"]:
+                        log.info(f"{snap_name} Created and ready to use")
+                        snap_con_name = res["status"]["boundVolumeSnapshotContentName"]
+                        break
+                    else:
+                        log.info(
+                            f"{snap_name} is not ready yet, sleep {sleep_time} sec before re-check"
+                        )
+                        time.sleep(sleep_time)
+                        timeout -= sleep_time
+                except Exception:
+                    log.info(
+                        f"{snap_name} is not ready yet, sleep {sleep_time} sec before re-check"
+                    )
+                    time.sleep(sleep_time)
+                    timeout -= sleep_time
+
+            else:
+                err_msg = f"Can not get snapshot status {res}"
+                log.error(err_msg)
+                raise Exception(err_msg)
+
+        return self.get_creation_time(snap_name, snap_con_name, UTC_datetime)
+
+    def read_logs(self, kind, namespace, start_time):
```

Comment:
same here, good to move it to common library

Reply:
same as above.